### DISCUSSION TITLE
How to set to show the full range of the view, and about the timezone

### DISCUSSION THREAD

**Discussion by TigerQi** [highfestiva/finplot#490] [Q&A]
Hi, first, thanks and finplot has the great performance.

question1, by default, the view only show part of the full range of the data, how to set to show the full range at start. 
below to create the plotwidget:
        self.plot_widget= fplt.create_plot_widget(master=window, rows=3, init_zoom_periods=100)
I try use:
        vb.v_zoom_scale = 1
        fplt.refresh()
but seem not work, its still show the default part of the range. thanks

question2, I refered the example analyze.py to prepare the data, but I found the date showd on the x-axis has 3 hours gap. I use the data for china maket, the close time is 3pm, but the lable shows is 6pm
I use it like:
fplt.timestamp_format = '%Y-%M-%D %H:%M'
fplt.timezone = timedelta(hours=+8)

and below to make the origianl data to what finplot use:
        d['time'] = d['trade_date'].apply(lambda x: int(x.timestamp()))
        df['date']=d['time']
        df = pd.concat([df, d['time']], axis=1)
        df = df.set_index('time').sort_index(axis=0, ascending=True, kind='mergesort')

Thank!

**Comment by TigerQi** [comment]
![微信截图_20240120232825](https://github.com/highfestiva/finplot/assets/71312398/b033e99a-57b4-4951-8abb-7136ebf6c936)

I use set_x_pos and checked the min, max is correct in the df, still not work

**Comment by TigerQi** [comment]
for timezone; I now use fplt.display_timezone = gettz('CCT'); it works; use fplt.display_timezone = gettz('Asia/Shanghai') will not work

**Reply by Protomolekule** [reply]
Maybe yo need to localize your datetime first.
```
 if df.index.tz is None:
                    df.index = pd.to_datetime (df.index)
                    df.index = df.index.tz_localize ('CCT')
```
or you need to convert it:

`df.index = df.index.tz_convert ('Asia/Shanghai')`

depending on the source of your data.

**Comment by TigerQi** [comment]
for set_x_pos, I found if add a button click to run, it works, but if not, no matter where I put the code, it not work.

**Comment by highfestiva** [comment]
This sets the number of candles shown to 10,000,000,000:
```
ax = fplt.create_plot('My Title', init_zoom_periods=1e10)
```
